roles/awx/tasks/install.yml

```diff
@@ -0,0 +1,105 @@
+---
+# Based on https://github.com/ansible/awx/blob/1.0.3/installer/openshift/tasks/main.yml
+
+- name: Ensure AWX Project Exists
+  oc_project:
+    name: "{{ awx_project }}"
+
+# This secret is created by postgresql-persistent template
+- name: Read Existing Postgres Secret
+  oc_secret:
+    state: list
+    namespace: "{{ awx_project }}"
+    name: postgresql
+    decode: true
+  register: awx_postgres_secret
+  no_log: true
+
+- name: Get or Generate AWX Postgres Password
+  set_fact:
+    pg_password: "{{ awx_postgres_secret.results.decoded['database-password'] | default(16 | lib_utils_oo_random_word) }}"
+  no_log: true
+
+- name: Deploy and Activate Postgres
+  # Note: NAMESPACE param is where to look for postgresql ImageStream.
+  shell: >-
+    {{ openshift_client_binary }} project {{ awx_project }};
```

Comment:
I'd breakout namespace creation into a discrete task

Reply:
This did not create the project, only switched to it to control where `oc apply` instantiates (this template omits `namespace:`).
=> Discovered I can use the cleaner `oc apply -n ...`, switched to that.